sdk/storage/storage-file-datalake/src/Pipeline.ts

```diff
@@ -0,0 +1,442 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  CompatResponse as HttpOperationResponse,
+  RequestPolicy as IHttpClient,
+  HttpHeadersLike as HttpHeaders,
+  RequestPolicy,
+  RequestPolicyFactory,
+  RequestPolicyOptionsLike as RequestPolicyOptions,
+  WebResourceLike as WebResource,
+  KeepAliveOptions,
+  ExtendedServiceClientOptions,
+  convertHttpClient,
+  createRequestPolicyFactoryPolicy,
+  HttpPipelineLogLevel,
+} from "@azure/core-http-compat";
+import {
+  RequestBodyType as HttpRequestBody,
+  ProxySettings as ProxyOptions,
+  UserAgentPolicyOptions as UserAgentOptions,
+  bearerTokenAuthenticationPolicy,
+  Pipeline as CorePipeline,
+  decompressResponsePolicyName,
+  PipelinePolicy,
+  HttpClient,
+} from "@azure/core-rest-pipeline";
+import { authorizeRequestOnTenantChallenge, createClientPipeline } from "@azure/core-client";
+import { parseXML, stringifyXML } from "@azure/core-xml";
+import { TokenCredential, isTokenCredential } from "@azure/core-auth";
+
+import { logger } from "./log";
+import { StorageRetryOptions, StorageRetryPolicyFactory } from "../../storage-blob/src/StorageRetryPolicyFactory";
```

Comment:
The files are in blob folder, but seems used in all service B/Q/F. 
Should we put it to a common folder?